allennlp/tests/common/test_util.py

```diff
@@ -88,3 +103,30 @@ def test_get_frozen_and_tunable_parameter_names(self):
         ) = util.get_frozen_and_tunable_parameter_names(model)
         assert set(frozen_parameter_names) == {"linear.weight", "linear.bias"}
         assert set(tunable_parameter_names) == {"conv.weight", "conv.bias"}
+
+    def test_plugins_are_discovered_and_imported(self):
+        plugins_root = self.FIXTURES_ROOT / "plugins"
+        project_a_fixtures_root = plugins_root / "project_a"
+        project_b_fixtures_root = plugins_root / "project_b"
+
+        with tempfile.TemporaryDirectory() as temp_dir_a, tempfile.TemporaryDirectory() as temp_dir_b:
+            distutils.dir_util.copy_tree(project_a_fixtures_root, temp_dir_a)
+            distutils.dir_util.copy_tree(project_b_fixtures_root, temp_dir_b)
+
+            # We make one plugin available in the path, in another directory as if it were another project.
+            sys.path.append(temp_dir_a)
+
+            # We move to another directory with a different plugin, as if it were other project.
+            with pushd(temp_dir_b):
+                # In general when we run scripts or commands in a project, the current directory is the root of it
+                # and is part of the path. So we emulate this here.
+                sys.path.append(".")
+
+                available_plugins = list(util.discover_plugins())
```

Comment:
I don't understand why we need these two different methods.

Reply:
There's no need. In case someone wants to see the available plugins and check if a particular one is available. I can merge them anyway.